docs/api-reference/core/transform/buffer-transform.md

```diff
@@ -0,0 +1,78 @@
+# BufferTransform
+
+`BufferTransform` is an internal helper class for `Transform`, responsible for managing resources and state required for reading from and/or writing to `Buffer` objects. It auto creates `feedbackBufferes` when requested, creates `TransformFeedback` objects. Maintains all buffer bindings, when swapping is eanbled, two binding objects are created for easy switching of all WebGL resource binginds.
+
+NOTE: In following sections 'buffer transform' is used to refer to 'reading from and/or writing to `Buffer` objects'.
+
+## Constructor
+
+### Transform(gl : WebGL2RenderingContext, props: Object)
+
+* `gl` (`WebGLRenderingContext`) gl - context
+* `props` (`Object`, Optional) - contains following data.
+  * `sourceBuffers` (`Object`, Optional) - key and value pairs, where key is the name of vertex shader attribute and value is the corresponding `Attribute`, `Buffer` or attribute descriptor object.
+  * `feedbackBuffers` (`Object`, Optional) - key and value pairs, where key is the name of vertex shader varying and value is the corresponding `Buffer` object or buffer params object. If a buffer params object is specified, it will contain following fields, these can be used to capture data into the buffer at particular offset and size.
+    * `buffer`=(Buffer) - Buffer object to be bound.
+    * `byteOffset`=(Number, default: 0) - Byte offset that is used to start recording the data in the buffer.
+    * `byteSize`=(Number, default: remaining buffer size) - Size in bytes that is used for recording the data.
```

Comment:
Maybe not for this PR: Can we follow the same accessor convention that `model.setAttributes` uses?